RELEASE.rst

```diff
@@ -45,32 +45,17 @@ and a `source distribution <https://packaging.python.org/guides/distributing-pac
      $ git push
      $ git push --tags
 
-#. Build the release packages by running the ``release.sh``
-   script on macOS::
-
-     $ git clone git@github.com:mongodb/motor.git
-     $ cd motor
-     $ git checkout "<release version number>"
-     $ ./release.sh
-
-   This will create the following distributions::
-
-     $ ls dist
-     motor-<version>.tar.gz
-     motor-<version>-py3-none-any.whl
-
-#. Upload all the release packages to PyPI with twine::
+#. Bump the version number to ``<next version>.dev0`` in ``motor/_version.py``,
+   commit, push.
 
-     $ python3 -m twine upload dist/*
+#. Authorize the deployment for the tagged version on the release GitHub Action and
+   wait for it to successfully publish to PyPI.
 
 #. Make sure the new version appears on https://motor.readthedocs.io/. If the
    new version does not show up automatically, trigger a rebuild of "latest":
    https://readthedocs.org/projects/motor/builds/
 
-#. Bump the version number to <next version>.dev0 in motor/_version.py,
-   commit, push.
-
-#. Publish the release version in Jira.
+#. Publish the release version in Jira and add a description.
```

Comment:
It would be helpful to briefly outline what the description should contain.

Reply:
Done